functions/venue.js

```diff
@@ -593,11 +597,7 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
     throw new HttpsError("not-found", `Venue ${venueId} not found`);
   }
 
-  // @debt this is exactly the same as in updateVenue_v2
-  const updated = doc.data();
-
-  // @debt refactor function so it doesn't mutate the passed in updated object, but efficiently returns an updated one instead
-  createBaseUpdateVenueData(data, updated);
+  const updated = createBaseUpdateVenueData(data, doc);
```

Comment:
not sure if that should've been changed.

`data` object is mutated as the `@debt` comment says, and that part should be refactored anyway. and if it didn't return a value, it'd be easier to read that I believe.
what do you think?

Reply:
> so it doesn't mutate the passed in **updated** object

I think he meant the variable with the name `updated`. The code inside `createBaseUpdateVenueData` does not mutate the `data` object.

Reply:
ah, i see. Looks like you're right. 
okay, let's have it as you did then.